modules/standard/GPU.chpl

```diff
@@ -626,4 +626,191 @@ module GPU
   */
   inline proc gpuMaxLocReduce(const ref A: [] ?t) do return doGpuReduce("maxloc", A);
 
+
+
+  // ============================
+  // GPU Scans
+  // ============================
+
+  // The following functions are used to implement GPU scans. They are
+  // intended to be called from a GPU locale.
+  import BitOps;
+  import Math;
+
+
+  proc gpuScan(ref gpuArr: [] uint){
```

Comment:
I guess as long as the type supposed `+` there isn't a reason to not support it at this point. It was originally `uint` becuase that's what radix sort supports right now, but now that this is a standalone feature, we can make it generic.

Reply:
Ok, I was wondering if there's something I was missing.

Hopefully it's just a matter of removing or replacing the type with `[] ?t` (and I guess adding `where isNumericType(t) && !isComplexType(t)`).

If there's some other assumptions baked in that makes that difficult, I'd be fine having this PR proceed "as is" with broadening it as a followup task.

If you do make it generic, it would also be good to add signed int and real version of the tests as well.